src/Data/Enum.purs

```diff
@@ -242,3 +242,10 @@ defaultToEnum n
   -- | Runs in `O(n)` where `n` is `fromEnum a`
 defaultFromEnum :: forall a. Enum a => a -> Int
 defaultFromEnum = maybe 0 (\prd -> defaultFromEnum prd + 1) <<< pred
+
+-- | Like `toEnum` but returns `bottom` if `x` is less than `fromEnum bottom`
+-- | and `top` if `x` is greater than `fromEnum top`.
+toBoundedEnum :: forall a. (BoundedEnum a) => Int -> a
+toBoundedEnum x = case toEnum x of
+  Just enum -> enum
+  Nothing -> if x < fromEnum (bottom :: a) then bottom else top
```

Comment:
It's probably cheaper to make the check here before the call to `toEnum`, especially when the enum is built out of things like `Either`.